javascript/code-challenges/challenges-07.test.js

```diff
@@ -0,0 +1,365 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called sortStarWarsCharacters that sorts the characters in the starWarsPeople array by height from tallest to shortest.
+------------------------------------------------------------------------------------------------ */
+
+let starWarsPeople = [
+  {
+    "name": "C-3PO",
+    "height": "167",
+    "eye_color": "yellow"
+  },
+  {
+    "name": "Luke Skywalker",
+    "height": "172",
+    "eye_color": "blue"
+  },
+  {
+    "name": "R2-D2",
+    "height": "96",
+    "eye_color": "red"
+  }
+];
+
+const sortStarWarsCharacters = (starWarsArr) => {
+  return starWarsArr.sort((a, b) => b.height - a.height);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named removeThree that takes an index and an array. The function should removes three items in the array starting with the value at the index. 
+------------------------------------------------------------------------------------------------ */
+
+const removeThree = (idx, arr) => {
+  arr.splice(idx, 3);
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named joinArray that takes an array and joins all of the elements together in one string on a space.
+------------------------------------------------------------------------------------------------ */
+
+const joinArray = (arr) => {
+  return arr.join(' ');
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named howMuchPencil that takes in a string, as written on the side of a pencil.
+
+As you sharpen the pencil, the string will become shorter and shorter, starting by removing the first letter.
+
+Your function should use slice within a loop and return an array of each successive string result from losing letters to the sharpener, until nothing is left.
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  let numOfletters = str.split('');
```

Comment:
This is unnecessary because the **.length** function works on strings too! You don't need to create an array.